src/SysInfo.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2023 - Ed Beroset <beroset@ieee.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef SYSINFO_H
+#define SYSINFO_H
+
+#include <sys/sysinfo.h>
+
+#include <QObject>
+#include <QtQml>
+
+class SysInfo : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(long uptime READ uptime NOTIFY uptimeChanged)
+    Q_PROPERTY(QList<qreal> loads READ loads NOTIFY loadsChanged)
+    Q_PROPERTY(unsigned long totalram READ totalram NOTIFY totalramChanged)
+    Q_PROPERTY(unsigned long freeram READ freeram NOTIFY freeramChanged)
+    Q_PROPERTY(unsigned long sharedram READ sharedram NOTIFY sharedramChanged)
+    Q_PROPERTY(unsigned long bufferram READ bufferram NOTIFY bufferramChanged)
+    Q_PROPERTY(unsigned long totalswap READ totalswap NOTIFY totalswapChanged)
+    Q_PROPERTY(unsigned long freeswap READ freeswap NOTIFY freeswapChanged)
+    Q_PROPERTY(unsigned procs READ procs NOTIFY procsChanged)
+    Q_PROPERTY(unsigned long totalhigh READ totalhigh NOTIFY totalhighChanged)
+    Q_PROPERTY(unsigned long freehigh READ freehigh NOTIFY freehighChanged)
+    Q_PROPERTY(unsigned mem_unit READ mem_unit CONSTANT)
```

Comment:
These Q_PROPERTY are meant to be exposed to QML so they should follow the QML coding style where fields are in snakeCase, so this should be memUnit to keep the QML code tidy

Reply:
Ok